scripts/ci-helper.sh

```diff
@@ -163,48 +83,56 @@ function build() {
     ldflags="-s -w "
   fi
 
-  ldflags="${ldflags}-X github.com/hashicorp/vault/version.Version=$version -X github.com/hashicorp/vault/version.GitCommit=$revision -X github.com/hashicorp/vault/version.BuildDate=$build_date"
+  # if building locally with enos - don't need to set version/prerelease/metadata as the default from version_base.go will be used
+  ldflags="${ldflags} -X github.com/hashicorp/vault/version.GitCommit=$revision -X github.com/hashicorp/vault/version.BuildDate=$build_date"
+
+  if [[ ${BASE_VERSION+x} ]]; then
+    msg="${msg}, base version ${BASE_VERSION}"
+    ldflags="${ldflags} -X github.com/hashicorp/vault/version.Version=$BASE_VERSION"
+  fi
+
+  if [[ ${PRERELEASE_VERSION+x} ]]; then
+    msg="${msg}, prerelease ${PRERELEASE_VERSION}"
+    ldflags="${ldflags} -X github.com/hashicorp/vault/version.VersionPrerelease=$PRERELEASE_VERSION"
+  fi
 
-  if [ -n "$prerelease" ]; then
-    msg="${msg}, prerelease ${prerelease}"
-    ldflags="${ldflags} -X github.com/hashicorp/vault/version.VersionPrerelease=$prerelease"
+  if [[ ${VERSION_METADATA+x} ]]; then
+    msg="${msg}, metadata ${VERSION_METADATA}"
+    ldflags="${ldflags} -X github.com/hashicorp/vault/version.VersionMetadata=$VERSION_METADATA"
   fi
 
-  if [ -n "$metadata" ]; then
-    msg="${msg}, metadata ${metadata}"
-    ldflags="${ldflags} -X github.com/hashicorp/vault/version.VersionMetadata=$metadata"
+  if [ ${GO_TOOLCHAIN} == "fips" ]; then
+    mkdir .tools
+    cd .tools
+    curl --show-error --silent --fail --user "${ARTIFACTORY_USER}:${ARTIFACTORY_TOKEN}" --request GET \
+    "${ARTIFACTORY_ENDPOINT}/${ARTIFACTORY_SOURCE_REPO}/fips-golang-go/${FIPS_GO_VERSION}/${FIPS_GO_COMMIT}/${FIPS_GO_ARTIFACT}" \
+    --output go.tar.gz
+    echo "$FIPS_GO_CHECKSUM  go.tar.gz" | sha256sum -c -
+    sudo apt update && sudo apt install -y clang
+    mkdir go && tar --strip-components=2 -C go -xf go.tar.gz
+    export GOROOT=$(pwd)/go
+    echo $PATH
+    pwd
+    export PATH=$(pwd)/go/bin:${PATH}
+    echo $PATH
+    export GOEXPERIMENT="boringcrypto"
+    $(pwd)/go/bin/go env
+    cd ..
   fi
 
   # Build vault
   echo "$msg"
-  pushd "$(repo_root)"
-  mkdir -p dist
-  mkdir -p out
-  set -x
-  go build -v -tags "$GO_TAGS" -ldflags "$ldflags" -o dist/
-  set +x
-  popd
-}
-
-# Bundle the dist directory into a zip
-function bundle() {
-  : "${BUNDLE_PATH:=$(repo_root)/vault.zip}"
-  echo "--> Bundling dist/* to $BUNDLE_PATH"
-  zip -r -j "$BUNDLE_PATH" dist/
+  go build -o "$BIN_PATH" -tags "$GO_TAGS" -ldflags "$ldflags" -trimpath -buildvcs=false
 }
 
 # Prepare legal requirements for packaging
 function prepare_legal() {
   : "${PKG_NAME:="vault"}"
-
-  pushd "$(repo_root)"
-  mkdir -p dist
```

Comment:
I wasn't able to find where we're now creating the `dist` dir, any points?

Reply:
I guess it's implicitly created by the go build command as a result of us using `-o $BIN_PATH` and defaulting $BIN_PATH to `dist/`.  Maybe we should use BIN_PATH more widely instead of hardcoding `dist` here?

Reply:
Yes we do use `$BIN_PATH` instead of hard-coding dist I think. [Here](https://github.com/hashicorp/vault/blob/7a227fb061295bfcb7e81484020464c5981d02a4/scripts/ci-helper.sh#L71) is how we handle it in the `build` function. LMK if you had other thoughts around this though